### PR TITLE
Delta subscription handlers

### PR DESCRIPTION
**Pull request by peterschwarz**
Add the protobuf messages for State Delta Export subscribe and
unsubscribe.  This includes requests and responses, as well as the
message codes for the values.

Adds the `StateDeltaProcessor` to manage subscribers for state deltas.
Introduces a pair of handlers for registering and unregistering the
subscibers.

Subscribers are rejected if their registration request does not have a
block id that is in the current chain.

### CODE REVIEW

File: protos/state_delta.proto [hunk 1]
```diff
@@ -36,3 +36,53 @@ message StateChange {
...
+message RegisterStateDeltaSubscriberResponse {
+    enum Status {
+        // returned on successful registration
+        OK = 0;
+        // returned on a failed registration, due to
+        // an internal validator error
+        INTERNAL_ERROR = 1;
+        // returned on a failed registration, due to the
+        // last_known_block_id being unknown.  This could imply
+        // that a fork had a occurred and been resolved since
```
Comment (boydjohnson) [inline protos/state_delta.proto]: 'had a occurred' should be 'had occurred'.

File: validator/sawtooth_validator/state/state_delta_processor.py [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+        with self._subscriber_cond:
+            return list(self._subscribers.keys())
+
+    def add_subscriber(
+        self,
+        connection_id,
+        last_known_block_ids,
+        address_prefix_filters
+    ):
+        """Add a subscriber for deltas over specific address_prefixs.
```
Comment (boydjohnson) [inline validator/sawtooth_validator/state/state_delta_processor.py]: address_prefixs -> address_prefixes.

File: validator/sawtooth_validator/state/state_delta_processor.py [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+        Args:
+            connection_id (str): a connection id
+            last_known_block_ids (list of str): a list of block ids known to
+                to the subscriber, expected in order of newest to oldest
+            address_prefix_filters (list of str): a list of address prefixes
+                to filter
+
+        Raises:
+            NoKnownBlockError: if the list of last_known_block_ids does not
+                contain a block id contianed in the block store.
```
Comment (boydjohnson) [inline validator/sawtooth_validator/state/state_delta_processor.py]: contianed -> contained.

File: validator/sawtooth_validator/state/state_delta_processor.py [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+    _msg_type = validator_pb2.Message.STATE_DELTA_SUBSCRIBE_RESPONSE
+
+    def __init__(self, delta_processor):
+        self._delta_processor = delta_processor
+
+    def handle(self, connection_id, message_content):
+        request = RegisterStateDeltaSubscriberRequest()
+        request.ParseFromString(message_content)
+
+        ack = RegisterStateDeltaSubscriberResponse()
```
Comment (delventhalz) [inline validator/sawtooth_validator/state/state_delta_processor.py]: Acknowledgement? Is this a commonly understood abbreviation?

Comment (peterschwarz) [inline validator/sawtooth_validator/state/state_delta_processor.py]: We use this in all of the interconnect handlers.

File: validator/sawtooth_validator/state/state_delta_processor.py [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+    """The StateDeltaProcessor manages subscribers for state deltas.
+    """
+
+    def __init__(self, service, state_delta_store, block_store):
+        """Constructs a new StateDeltaProcessor.
+
+        Args:
+            service (:obj:`Interconnect`): The zmq internal interface.
+            state_delta_store (:obj:`StateDeltaStore`): The source of state
+                delta values.
```
Comment (peterschwarz) [inline validator/sawtooth_validator/state/state_delta_processor.py]: done

**Pull request merged**